### ISSUE TITLE
FusedLinear not recorded by GradientTape — zero gradients for all DenseLayer training

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet.Tensors#102]
## Bug Description

\`CpuEngine.FusedLinear<T>()\` performs \`output = input @ weights + bias\` but does NOT record the operation on the active \`GradientTape\`. When \`TrainWithTape()\` computes gradients via \`tape.ComputeGradients(loss, parameters)\`, the tape has no record of how the parameters were used in the forward pass, so all parameter gradients are zero.

This affects **every neural network** that uses DenseLayer, FullyConnectedLayer, or any layer whose Forward calls \`Engine.FusedLinear\`.

## Reproduction

\`\`\`csharp
var network = new FeedForwardNeuralNetwork<double>(...);
var input = new Tensor<double>([4]);
var target = new Tensor<double>([1]);

var paramsBefore = network.GetParameters().ToArray();
network.Train(input, target);  // Uses TrainWithTape internally
var paramsAfter = network.GetParameters().ToArray();

// paramsBefore == paramsAfter — parameters didn't change!
// Because gradients were all zero.
\`\`\`

## Root Cause

\`DenseLayer.Forward()\` calls \`Engine.FusedLinear(input, _weights, _biases, activation)\` at line ~846. \`FusedLinear\` in CpuEngine (line ~19289) performs the matmul + bias + activation directly on raw arrays via \`CpuFusedOperations.FusedGemmBiasActivation\`, completely bypassing the tape.

For comparison, \`TensorMatMul\` is also not tape-recorded, but the tape is designed to record operations at the tensor level. The issue is that **none** of the engine's compute methods (\`FusedLinear\`, \`TensorMatMul\`, \`TensorBroadcastAdd\`, etc.) register themselves with the active \`GradientTape\`.

## Expected Behavior

When a \`GradientTape\` scope is active, engine operations that involve trainable parameter tensors should be recorded so that \`tape.ComputeGradients()\` can trace the dependency chain from loss → output → parameters.

## Proposed Fix

**Option A: Record FusedLinear as a single tape op**

When a tape is active, \`FusedLinear\` should record itself as a composite operation with known analytical gradients:
- \`dL/dW = input^T @ dL/doutput\`
- \`dL/db = sum(dL/doutput, axis=0)\`
- \`dL/dinput = dL/doutput @ W^T\`

**Option B: Decompose FusedLinear into tape-tracked primitives**

When a tape is active, \`FusedLinear\` falls back to \`TensorMatMul + TensorAdd\` which are individually tape-recorded. This is simpler but loses the fusion optimization during training (acceptable since training is forward+backward anyway).

## Impact

- RecurrentNeuralNetwork: 4/17 failures
- Autoencoder: 2/17 failures  
- HopeNetwork: 6/17 failures
- DeepQNetwork: 3/17 failures
- Any network using DenseLayer with TrainWithTape

## Separate from #100

Issue #100 is about FusedLinear using naive loops instead of BLAS for double (performance). This issue is about FusedLinear not being tape-tracked at all (correctness). Both need fixing but are independent.
